examples/chart/teleport-cluster/charts/teleport-operator/templates/_helpers.tpl

```diff
@@ -75,3 +76,56 @@ teleport.dev/majorVersion: '{{ include "teleport-cluster.majorVersion" . }}'
 {{- /* This template is a placeholder.
 If we are imported by the main chart "teleport-cluster" it is overridden*/ -}}
 {{- define "teleport-cluster.auth.serviceFQDN" -}}{{- end }}
+
+{{- /* This templates returns "true" or "false"  describing if the CRDs should be deployed.
+If we have an explicit requirement ("always" or "never") things are easy.
+If we don't we check if the operator is enabled.
+However, we cannot just trash the CRDs if the operator is disabled, this causes
+a mass CR deletion and users will shoot themselves in the foot whith this
+(temporarily disabling the operator would cause havoc).
+So we check if there's a CRD already deployed, it that's the case, we keep the CRDs.
+*/ -}}
+{{- define "teleport-cluster.operator.shouldInstallCRDs" -}}
+  {{- if eq .Values.installCRDs "always" -}}
+    true
+  {{- else if eq .Values.installCRDs "never" -}}
+    false
+  {{- else if eq .Values.installCRDs "dynamic" -}}
+    {{- if .Values.enabled -}}
+      true
+    {{- else -}}
+      {{- include "teleport-cluster.operator.checkExistingCRDs" . -}}
+    {{- end -}}
+  {{- else -}}
+    {{- fail ".Values.installCRDs must be 'never', 'always' or 'dynamic'." -}}
+  {{- end -}}
+{{- end -}}
+
+{{- /* This template checks if a known CRD is depployed (rolev7) and owned by
+the release. As CRDs are not namespaced, we must use a custom annotation to avoid
+a conflict when two releases are deployed with the same name in different namespaces. */ -}}
+{{- define "teleport-cluster.operator.checkExistingCRDs" -}}
+  {{ $existingCRD := lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "teleportrolesv7.resources.teleport.dev"}}
```

Comment:
Should we check for a resource that wasn't introduced in teleport v15?

Reply:
I would say the opposite, we want to check for a new resource that will not get removed.
I don't think this change can be backported as it's too large for a bugfix and adds new features. This might land in 15.2.